src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/TripleDES/TripleDESCipherTests.cs

```diff
@@ -23,19 +20,45 @@ public static void TripleDESDefaults()
         }
 
         [Fact]
-        public static void TripleDESRoundTrip192BitsNoneECB()
+        public static void TripleDESGenerate128Key()
         {
-            byte[] key = "c5629363d957054eba793093b83739bb78711db221a82379".HexToByteArray();
+            using (TripleDES des = TripleDESFactory.Create())
+            {
+                des.KeySize = 128;
+                byte[] key = des.Key;
+                Assert.Equal(128, key.Length * 8);
+            }
+        }
+
+        [Fact]
+        public static void TripleDESInvalidKeySizes()
+        {
+            using (TripleDES des = TripleDESFactory.Create())
+            {
+                Assert.Throws<CryptographicException>(() => des.KeySize = 128 - 1);
+                Assert.Throws<CryptographicException>(() => des.KeySize = 192 + 1);
+            }
+        }
+
+        [Theory]
+        [InlineData(192, "c5629363d957054eba793093b83739bb78711db221a82379", "e56f72478c7479d169d54c0548b744af5b53efb1cdd26037")]
+        [InlineData(128, "c5629363d957054eba793093b83739bb",                 "1387b981dbb40f34b915c4ed89fd681a740d3b4869c0b575")]
+        [InlineData(192, "c5629363d957054eba793093b83739bbc5629363d957054e", "1387b981dbb40f34b915c4ed89fd681a740d3b4869c0b575")]
+        public static void TripleDESRoundTripNoneECB(int keySize, string keyHex, string expectedCipherHex)
```

Comment:
Two comments about these:
1) `keySize` is always derivable from `keyHex`
2) The formatting here looks good, but won't stand up to auto-reformatting tools.  Perhaps flip the key and expectedValue so that the formatting isn't so fragile?